jans-cedarling/cedarling/src/common/policy_store.rs

```diff
@@ -5,155 +5,430 @@
 
-        #[test]
-        fn test_string_error() {
-            static POLICY_STORE_RAW: &str =
-                include_str!("../../../test_files/policy-store_policy_err_broken_utf8.json");
+        // TODO: this isn't really a human readable format so idk why the error message is
+        // like this. Look into this in the future.
+        assert_eq!(err_msg, "unable to decode policy with id: 840da5d85403f35ea76519ed1a18a33989f855bf1cf8, error: unable to decode policy_content from human readable format: unexpected token `)` at line 9 column 5")
+    }
 
-            let policy_result = serde_json::from_str::<PolicyStoreMap>(POLICY_STORE_RAW);
-            assert!(policy_result
-                .unwrap_err()
-                .to_string()
-                .contains(&ParsePolicySetMessage::String.to_string()));
-        }
+    /// Tests that a valid version string is accepted.
+    #[test]
+    fn test_valid_version() {
+        let valid_version = "1.2.3".to_string();
+        assert!(check_cedar_version(serde_json::Value::String(valid_version)).is_ok());
+    }
 
-        #[test]
-        fn test_policy_error() {
-            static POLICY_STORE_RAW: &str =
-                include_str!("../../../test_files/policy-store_policy_err_broken_policy.json");
+    /// Tests that a valid version string with 'v' prefix is accepted.
+    #[test]
+    fn test_valid_version_with_v() {
+        let valid_version_with_v = "v1.2.3".to_string();
+        assert!(check_cedar_version(serde_json::Value::String(valid_version_with_v)).is_ok());
+    }
 
-            let policy_result = serde_json::from_str::<PolicyStoreMap>(POLICY_STORE_RAW);
-            let err_msg = policy_result.unwrap_err().to_string();
-            assert_eq!(err_msg,"unable to decode policy with id: 840da5d85403f35ea76519ed1a18a33989f855bf1cf8, error: unable to decode policy_content from human readable format: unexpected token `)` at line 15 column 1")
-        }
+    /// Tests that an invalid version format is rejected.
+    #[test]
+    fn test_invalid_version_format() {
+        let invalid_version = "1.2".to_string();
+        assert!(check_cedar_version(serde_json::Value::String(invalid_version)).is_err());
+    }
+
+    /// Tests that an invalid version part (non-numeric) is rejected.
+    #[test]
+    fn test_invalid_version_part() {
+        let invalid_version = "1.two.3".to_string();
+        assert!(check_cedar_version(serde_json::Value::String(invalid_version)).is_err());
+    }
+
+    /// Tests that an invalid version format with 'v' prefix is rejected.
+    #[test]
+    fn test_invalid_version_format_with_v() {
+        let invalid_version_with_v = "v1.2".to_string();
+        assert!(check_cedar_version(serde_json::Value::String(invalid_version_with_v)).is_err());
+    }
+
+    /// Tests that an error is returned for multiple role mappings in token metadata.
+    #[test]
+    fn test_invalid_multiple_role_mappings_in_token_metadata() {
```

Comment:
this code can be simplified to:

```suggestion
    #[test]
    fn test_invalid_multiple_role_mappings_in_token_metadata() {
        let token_metadata_value = serde_json::json!([
            { "type": "Access", "person_id": "aud" },
            { "type": "Id", "person_id": "sub", "role_mapping": "role" },
            { "type": "userinfo", "person_id": "email", "role_mapping": "role" }
        ]);

        assert!(
            check_token_metadata(token_metadata_value).is_err(),
            "expected an error for multiple role mappings"
        );
    }
```

Reply:
updated here: [33315bc](https://github.com/JanssenProject/jans/pull/9910/commits/33315bc3ca79af2aa4f23f33187b4b8ae49b29d4)